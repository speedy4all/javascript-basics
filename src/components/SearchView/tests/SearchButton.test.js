import { render, fireEvent } from '@testing-library/react'
import SearchButton from '../search-button';
import renderer from 'react-test-renderer';

const defaultProps = {
    onClick: jest.fn(),
    posts: [],
}

describe('SearchButton component', () => {
    it('should call onClick', () => {
        const { debug, getByTestId } = render(<SearchButton onClick={defaultProps.onClick} /> )
        debug();
        const button = getByTestId('clear-button')
        fireEvent.click(button);
        expect(defaultProps.onClick).toHaveBeenCalledTimes(1)
        fireEvent.click(button);
        fireEvent.click(button);
        fireEvent.click(button);
        fireEvent.click(button);
        expect(defaultProps.onClick).toHaveBeenCalledTimes(5)
        expect(button).toContainHTML('button')
    })

    it('should render the correct snapshot', () => {
        const tree = renderer
            .create(<SearchButton onClick={defaultProps.onClick} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
})
