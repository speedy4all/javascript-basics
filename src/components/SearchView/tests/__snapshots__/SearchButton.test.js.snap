// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchButton component should render the correct snapshot 1`] = `
<button
  data-testid="clear-button"
  onClick={[MockFunction]}
>
  Clear
</button>
`;
